internal/config/fw_base.go

```diff
@@ -28,21 +33,21 @@ func (r *RSCommon) Configure(ctx context.Context, req resource.ConfigureRequest,
 		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
 		return
 	}
-	r.client = client
+	r.Client = client
 }
 
 // DSCommon is used as an embedded struct for all framework data sources. Implements the following plugin-framework defined functions:
 // - Metadata
 // - Configure
 // client is left empty and populated by the framework when envoking Configure method.
-// dataSourceName must be defined when creating an instance of a data source.
+// DataSourceName must be defined when creating an instance of a data source.
 type DSCommon struct {
-	client         *MongoDBClient
-	dataSourceName string
+	Client         *MongoDBClient
```

Comment:
any chance we can leave this private and just access these throught methods? (same for `RSCommon`)

Reply:
coming from Java I see your point. But I think it's more Go idiomatic just to expose a field if it's going to be used, instead of having a get method that just returns the field. Anyway I don't have a strong opinion, we can do that if we think that's better.

Reply:
I am ok, I don't have a strong opinion either - whatever is more Go idiomatic I am fine.